misk/src/test/kotlin/misk/client/TypedHttpClientTest.kt

```diff
@@ -42,22 +48,64 @@ internal class TypedHttpClientTest {
   @Inject
   private lateinit var jetty: JettyService
 
-  private lateinit var client: ReturnADinosaur
+  @Inject
+  private lateinit var tracer: Tracer
+
+  private lateinit var clientInjector: Injector
+
+  private val dinosaurRequest = Dinosaur.Builder().name("trex").build()
 
   @BeforeEach
   fun createClient() {
-    val clientInjector = Guice.createInjector(MoshiModule(), ClientModule(jetty))
-    client = clientInjector.getInstance()
+    clientInjector =
+        Guice.createInjector(MockTracingBackendModule(), MoshiModule(), ClientModule(jetty))
   }
 
   @Test
   fun useTypedClient() {
-    val response = client.getDinosaur(Dinosaur.Builder().name("trex").build()).execute()
+    val client = clientInjector.getInstance(ReturnADinosaur::class.java)
+    val response = client.getDinosaur(dinosaurRequest).execute()
     assertThat(response.code()).isEqualTo(200)
     assertThat(response.body()).isNotNull()
     assertThat(response.body()?.name!!).isEqualTo("supertrex")
   }
 
+  @Test
+  fun embedsTrace() {
```

Comment:
We should also have a test that does multiple hops (from client to server to server again), to ensure that parent span ids propagate.